webservices/ph5tostationxml.py

```diff
@@ -495,6 +489,20 @@ def get_network_date(self):
                         
         return min_start_time, max_end_time
     
+    def trim_to_level(self, network):
+        if self.args.get('level').upper() == "NETWORK":
+            network.stations = []
+        elif self.args.get('level').upper() == "STATION":
+            # for station level show the selected_number_of_channels element
+            for station in network.stations:
+                station.selected_number_of_channels = 0
+                station.channels = []
+        elif self.args.get('level').upper() == "CHANNEL":
+            for station in network.stations:
+                for channel in station.channels:
+                    channel.response = None
+        return network
+ 

```

Comment:
This is not as efficient as checking the level in the main loop, but allows for all the trimming to be performed in one place which is more maintainable.